### ISSUE TITLE
How to disable tableview goes Upside when keyboard comes Up

### ISSUE THREAD

**Issue by patriksharma** [hackiftekhar/IQKeyboardManager#1488]
**Describe the bug**
I'm creating a chat view and when I click on textview, keyboard comes up. When i type a message and to display that message in tableview, I need to keep tableview in the frame but table view goes up and there is no way i found to put tableview Y axis down. 

### Versions
**Xcode:**  Xcode 10
**Mac OS:**  10.13.6
**Simulator/Device:** all devices
**Simulator/Device Name:** all devices
**Library Version:**

**Comment by AalokParikh** [comment]
> **Describe the bug**
> I'm creating a chat view and when I click on textview, keyboard comes up. When i type a message and to display that message in tableview, I need to keep tableview in the frame but table view goes up and there is no way i found to put tableview Y axis down.
> 
> ### Versions
> **Xcode:** Xcode 10
> **Mac OS:** 10.13.6
> **Simulator/Device:** all devices
> **Simulator/Device Name:** all devices
> **Library Version:**

I Got the same issue with Xcode 10.1
Did you find any workaround of this issue?

**Comment by AalokParikh** [comment]
> Yes @AalokParikh
> I fixed the issue. There is turnaround you need to do.
> 
> 1. First take all subview inside Scroll view. Keep text box outside scroll view.
> 2. Now you need to change position of Y of scrollview in textview did begin editing and end editing
> 
> ```
> func textViewDidBeginEditing(_ textView: UITextView) {
>        self.scrollView.setY(y: keyboardheight - 10)
> }
> ```
> Where keyboard height will we get from keyboard notification method.
> 
> ```
> func textViewDidEndEditing(_ textView: UITextView) {
>         self.scrollView.setY(y: 0)
> }
> ```

will you please tell what `keyboardheight` is? 
Where it is defined? or how to defined it?

**Comment by patriksharma** [comment]
add them in view did load 

    
  ```
      NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillShow(notification:)), name: NSNotification.Name.UIKeyboardWillShow, object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide(notification:)), name: NSNotification.Name.UIKeyboardWillHide, object: nil)
```
```

   @objc func keyboardWillShow(notification: Notification) {
        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue {
            keyboardheight = keyboardSize.height
            print("keyboard heigh",keyboardheight)
        }
    }
```

**Comment by AalokParikh** [comment]
> > Yes @AalokParikh
> > I fixed the issue. There is turnaround you need to do.
> > 
> > 1. First take all subview inside Scroll view. Keep text box outside scroll view.
> > 2. Now you need to change position of Y of scrollview in textview did begin editing and end editing
> > 
> > ```
> > func textViewDidBeginEditing(_ textView: UITextView) {
> >        self.scrollView.setY(y: keyboardheight - 10)
> > }
> > ```
> > Where keyboard height will we get from keyboard notification method.
> > ```
> > func textViewDidEndEditing(_ textView: UITextView) {
> >         self.scrollView.setY(y: 0)
> > }
> > ```
> 
> @patriksharma
> will you please tell what `keyboardheight` is?
> Where it is defined? or how to defined it?

@patriksharma 
Thanks for your quick reply but I'm not able to resolve the error relates to 

`self.scrollView.setY()` method 

it gives me error like ```Value of type 'UIScrollView?' has no member 'setY'```

**Comment by patriksharma** [comment]
@AalokParikh 

`self.scrollView.setY() ` I have created an seprate methd 

you can do like this 

`self.scrollView.frame.orgion.y `? .. got it ?

**Comment by hackiftekhar** [comment]
Btw, If you re working with a chat kind of screen then I prefer to disable library in that view controller and implement your own solution.

**Comment by patriksharma** [comment]
@hackiftekhar I don't agree with you. I did it in chat screen and kept IQKeyboardManager also. It is easy and not so difficult

All we need a scroll view and manage view Y position that's it. 

Library is useful as you don't need to manage to go tableview or collection view go up and down.

**Comment by hackiftekhar** [comment]
@partiksharma, I also agree with you that it’s not much difficult if you know that how to configure it and what view hierarchy needs to be in order to work. That’s why my statement about disabling library is a general statement for usual developers not for pro developers who know how to configure it :).
